src/main/webapp/resources/js/pages/projects/samples/components/LinkerModal.jsx

```diff
@@ -0,0 +1,104 @@
+import React from "react";
+import { Button, Checkbox, Form, Input, Modal, Space, Typography } from "antd";
+import styled from "styled-components";
+import { SPACE_SM } from "../../../../styles/spacing";
+import { BORDER_RADIUS, BORDERED_LIGHT } from "../../../../styles/borders";
+import { grey2 } from "../../../../styles/colors";
+import { LoadingOutlined } from "@ant-design/icons";
+import { getNGSLinkerCode } from "../../../../apis/linker/linker";
+
+const CommandText = styled(Typography.Paragraph)`
+  margin: 0px !important;
+  font-family: monospace;
+  font-size: 14px;
+`;
+
+const CommandWrapper = styled.div`
+  margin-top: ${SPACE_SM};
+  padding: 2px;
+  background-color: ${grey2};
+  border: ${BORDERED_LIGHT};
+  border-radius: ${BORDER_RADIUS};
+`;
+
+export default function LinkerModal({
+  visible,
+  sampleIds,
+  projectId,
+  onFinish,
+}) {
+  const [form] = Form.useForm();
+  const [scriptString, setScriptString] = React.useState();
+  const [command, setCommand] = React.useState();
+
+  const updateCommand = () => {
+    const types = form.getFieldValue("type");
+    setCommand(
+      types.length ? `${scriptString} -t ${types.join(",")}` : scriptString
+    );
+  };
+
+  const options = [
+    { label: i18n("Linker.fastq"), value: "fastq" },
+    { label: i18n("Linker.assembly"), value: "assembly" },
+  ];
+
+  React.useEffect(() => {
+    getNGSLinkerCode({ sampleIds, projectId }).then(({ data }) => {
+      // Post data to the server to get the linker command.
+      setScriptString(data);
+    });
+  }, [projectId, sampleIds]);
+
+  React.useEffect(updateCommand, [scriptString]);
+
+  return (
+    <Modal
+      className="t-linker-modal"
+      visible={visible}
+      title={i18n("Linker.title")}
+      onCancel={onFinish}
+      footer={
+        <div style={{ display: "flex", flexDirection: "row-reverse" }}>
+          <Button type="primary" onClick={onFinish}>
+            {i18n("Linker.close")}
+          </Button>
+        </div>
+      }
+    >
+      <Typography.Paragraph>{i18n("Linker.details")}</Typography.Paragraph>
+      <Typography.Text type="secondary">
+        <span dangerouslySetInnerHTML={{ __html: i18n("Linker.note") }} />
```

Comment:
Why do we need to use `dangerouslySetInnerHTML` here? Looking at `Linker.note` in the messages.properties file shows that there is no html in it

Reply:
Because I am asking React to set foreign HTML, not just a string.  It has to handle the parsing of that string to HTML and therefore needs to run it through a sanitizer.

Reply:
Whoops looks like I looked at the wrong line in the messages file. My bad :)